libyang/context.py

```diff
@@ -313,126 +313,85 @@ def parse_op_mem(
 
     def parse_data(
         self,
-        fmt: str,
         in_type: IOType,
         in_data: Union[str, bytes, IO],
-        parent: DNode = None,
-        parser_lyb_mod_update: bool = False,
-        parser_no_state: bool = False,
-        parser_parse_only: bool = False,
-        parser_opaq: bool = False,
-        parser_ordered: bool = False,
-        parser_strict: bool = False,
-        validation_no_state: bool = False,
-        validation_validate_present: bool = False,
-    ) -> Optional[DNode]:
+        fmt: str,
+        data: bool = False,
+        config: bool = False,
+        strict: bool = False,
+        trusted: bool = False,
+    ) -> DNode:
         if self.cdata is None:
             raise RuntimeError("context already destroyed")
-        parser_flgs = parser_flags(
-            lyb_mod_update=parser_lyb_mod_update,
-            no_state=parser_no_state,
-            parse_only=parser_parse_only,
-            opaq=parser_opaq,
-            ordered=parser_ordered,
-            strict=parser_strict,
-        )
-        validation_flgs = validation_flags(
-            no_state=validation_no_state, validate_present=validation_validate_present
+        parse_flags = parser_flags(
+            config=config,
+            strict=strict,
+            trusted=trusted,
         )
+
+        if data:
+            strict = False
+        validate_flags = validation_flags(config=config, strict=strict)
+
         fmt = data_format(fmt)
-        encode = True
         if fmt == lib.LYD_LYB:
-            encode = False
-        data = ffi.new("struct ly_in **")
-        data_keepalive = []
-        ret = data_load(in_type, in_data, data, data_keepalive, encode)
-        if ret != lib.LY_SUCCESS:
-            raise self.error("failed to read input data")
+            in_data = str2c(in_data, encode=False)
+        else:
+            in_data = str2c(in_data, encode=True)
 
-        if parent is not None:
-            ret = lib.lyd_parse_data(
-                self.cdata,
-                parent.cdata,
-                data[0],
-                fmt,
-                parser_flgs,
-                validation_flgs,
-                ffi.NULL,
+        dnode = ffi.new("struct lyd_node**")
+        if in_type == IOType.MEMORY:
+            ret = lib.lyd_parse_data_mem(
+                self.cdata, in_data, fmt, parse_flags, validate_flags, dnode
             )
-            lib.ly_in_free(data[0], 0)
-            if ret != lib.LY_SUCCESS:
-                raise self.error("failed to parse data tree")
-            return None
+        elif in_type == IOType.FD:
+            ret = lib.lyd_parse_data_fd(
+                self.cdata, in_data, fmt, parse_flags, validate_flags, dnode
+            )
```

Comment:
This is not related to "refactor v2 api to be more compatible with v1". Is it?

Reply:
indeed, I will create a new pr with only this update